modules/miner.go

```diff
@@ -5,68 +5,69 @@ import (
 	// corresponds to the header for 50 calls.
 	HeaderForWork() (types.BlockHeader, types.Target)
 
-	// MinerInfo returns a MinerInfo struct, containing information about the
-	// miner.
-	MinerInfo() MinerInfo
+	// SubmitBlock takes a block that has been worked on and has a valid
+	// target. Typically used with external miners.
+	SubmitBlock(types.Block) error
 
-	// MinerNotify is a channel to inform subscribers of when the miner has
-	// updated. This is primarily used for synchronization during testing.
-	MinerNotify() <-chan struct{}
+	// SubmitHeader takes a block header that has been worked on and has a
+	// valid target. A superior choice to SubmitBlock.
+	SubmitHeader(types.BlockHeader) error
+}
 
-	// SetThreads sets the number of threads in the miner.
-	SetThreads(int) error
+// A CPUMiner provides access to a single-threaded cpu miner.
+type CPUMiner interface {
+	// CPUHashrate returns the hashrate of the cpu miner in hashes per second.
+	CPUHashrate() int
 
-	// SolveBlock will have the miner make 1 attempt to solve the input block.
-	// It will give up after a few seconds, returning the block, a bool
-	// indicating whether it has been solved, and an error. SolveBlock is very
-	// slow, and will not submit the block the the consensus set.
-	SolveBlock(types.Block, types.Target) (types.Block, bool)
+	// Mining returns true if the cpu miner is enabled, and false otherwise.
+	CPUMining() bool
+
+	// FindBlock will have the miner make 1 attempt to find a solved block that
+	// builds on the current consensus set. It will give up after a few
+	// seconds, returning the block and a bool indicating whether the block is
+	// sovled.
+	FindBlock() (types.Block, bool)
 
 	// StartMining turns on the miner, which will endlessly work for new
 	// blocks.
-	StartMining() error
+	StartCPUMining()
 
 	// StopMining turns off the miner, but keeps the same number of threads.
-	StopMining() error
+	StopCPUMining()
 
-	// SubmitBlock takes a block that has been worked on and has a valid
-	// target. Typically used with external miners.
-	SubmitBlock(types.Block) error
+	// SolveBlock will have the miner make 1 attempt to solve the input block,
+	// which amounts to trying a few thousand different nonces. SolveBlock is
+	// primarily used for testing.
+	SolveBlock(types.Block, types.Target) (types.Block, bool)
+}
 
-	// SubmitHeader takes a block header that has been worked on and has a
-	// valid target. A superior choice to SubmitBlock.
-	SubmitHeader(types.BlockHeader) error
+// The Miner interface provides access to mining features.
+type Miner interface {
+	BlockManager
+	CPUMiner
+
+	// BlocksMined returns the number of blocks and stale blocks that have been
+	// mined using this miner.
+	BlocksMined() (goodBlocks, staleBlocks int)
+
+	// MinerNotify is a channel to inform subscribers of when the miner has
+	// updated. This is primarily used for synchronization during testing.
+	MinerNotify() <-chan struct{}
```

Comment:
We could put these two outer functions in each, and then the tester's would only need the CPUMiner and not the BlockManager, but I don't think it would make things that much neater.